### ISSUE TITLE
Schedule only on the selected days of the week not working on Send email campaign action after a decision

### ISSUE THREAD

**Issue by florentpetitjean** [mautic/mautic#7676]
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
When you schedule a send mail action in a campaign right after a decision, if you select some days you want the email sent and other not, it won't work the email will be sent no matter the day is allowed or not
Note that this happens only after a decision in campaign, it works fine if the action is not after a decision.

| Q   | A
| --- | ---
| Mautic version | 2.15.1
| PHP version | 
| Browser | Chrome

### Steps to reproduce
1. Create a test contact
2. Create a test segment with the contact in
3.Create a campaign starting with the segment
4. Make the campaign so an email is schedule right after a decision like this : 
![image](https://user-images.githubusercontent.com/49391402/60896223-40db7980-a266-11e9-8533-5835d7c6103a.png)
5. Schedule the second email with one day or more and select the days you want the emails send to (make sure that you don't allow the day when the email is supposed to be sent, so it will normaly be postponed) like this : 
![image](https://user-images.githubusercontent.com/49391402/60896383-8730d880-a266-11e9-9cba-6ecc53505710.png)
6. launch the campaign 
7. See in the contact history that the email is Schedule for the delay you set even if you don't allowed the day in the settings.
 
### Log errors

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

**Comment by moerazem** [comment]
I can confirm this issue on 2.15.2 as well.

**Comment by leroyagami** [comment]
I've similar problem here.
I created a campaign based on some actions to mark tag's and to send sequential e-mail's after the first e-mail readed(working like a trigger).

My e-mail's should be send fter the decision and scheduled of 2 by 2 days, or depending on the decision,to 4 by 4 days.

The problem is that they are scheduled all toghether at same day after the decision and at the same day rather to be sent one by one(always monday, wedensday ou friday)....

I'll show the image of my campaign to be clearer.
![Captura de tela de 2019-09-17 16-15-18](https://user-images.githubusercontent.com/21126849/65072234-dbdb8880-d966-11e9-9a87-dbf4ba74edd9.png)

**Comment by richmarks20** [comment]
**This also the case after a condition (as well as a decision).**

**Comment by incentfit** [comment]
I can confirm this is still happening in Mautic 2.15.3.  Here's a screenshot of the campaign and it's setup that was not respected.
![image](https://user-images.githubusercontent.com/13243272/68154647-f61cf600-ff15-11e9-8bbe-f0139864c229.png)

**Comment by oytuntez** [comment]
I think we are experiencing the same or similar bug, our contacts are not able to transition to the next step. In campaign event log, we only see the entry for the first email and decisions following it. No more email events after decisions.

**Comment by RCheesley** [comment]
Hi there, as we are only providing fixes for Mautic 3.x going forward, can you confirm if this continues in Mautic 3.0.1?
Thanks!

**Comment by ttencate** [comment]
I think we see this happening in Mautic 2.16.2, but only for leads whose timezone is NULL. Where the timezone is set (e.g. to Europe/Helsinki) the timezone also seems to affect the interval, which of course makes no sense.

We will test this on Mautic 3 soon, but I figured I'd record the above information because it might be useful in hunting down this bug (which probably still exists since M3 is mostly a big refactor, AIUI).
